src/CameraOnvif.cpp

```diff
@@ -0,0 +1,408 @@
+#include "CameraOnvif.hpp"
+
+#include <gsoap/plugin/wsseapi.h>
+#include <gsoap/plugin/wsddapi.h>
+#include "onvif/soapDeviceBindingProxy.h"
+#include "onvif/soapImagingBindingProxy.h"
+#include "onvif/soapMediaBindingProxy.h"
+#include "onvif/soapPTZBindingProxy.h"
+#include "onvif/soapPullPointSubscriptionBindingProxy.h"
+#include "onvif/soapRemoteDiscoveryBindingProxy.h"
+#include "wsdd.nsmap"
+
+#include <iostream>
+
+using namespace std;
+using namespace camera_onvif;
+
+struct CameraOnvif::Private {
+    struct soap *soap = soap_new1(SOAP_XML_STRICT | SOAP_XML_CANONICAL | SOAP_C_UTFSTRING);
+    MediaBindingProxy *proxy_media;
+    ImagingBindingProxy *proxy_image;
+    _trt__GetVideoEncoderConfigurationOptionsResponse *video_options;
+}defined;
+
+CameraOnvif::CameraOnvif(/* args */) {
+    user = "admin";
```

Comment:
This constructor definitely makes no sense. This is TW-specific. Today. For the camera you've set up.

Reply:
I was for fast test purpose only, I will remove it.